src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -516,6 +517,11 @@ private BoundLambda ReallyBind(NamedTypeSymbol delegateType)
 
         haveLambdaBodyAndBinders:
 
+            if (delegateType.IsExpressionTree())
+            {
+                ExpressionTreeRefLikeWalker.Analyze(binder.Compilation, lambdaSymbol, block, diagnostics);
```

Comment:
This isn't sufficient to catch parameters of ref struct types. Will need to analyze a different BoundNode (maybe just use the BoundLambda?)